ls1intum/Artemis
# `Communication`: Cleanup conversation after a course is deleted

### Checklist
#### General


#### Server

### Motivation and Context


Currently when we remove a course, that has conversations, we don't remove those from the table. 

### Description

This PR adds logic to remove the conversation of a removed course and also a migration, that will delete conversations of none existing courses. 

### Steps for Testing

Prerequisites:
- 1 Instructor

1. Since this PR introduces migration of deleting conversations of deleted courses, before running this branch or before deploying this PR delete a course with existing conversations and those conversations should include messages, replies and reactions. Then when you refresh the page you should get internal server error.
2. After you run this PR the database should be cleaned up and no internal server error should be displayed.
3. Then, Log in to Artemis
4. Create a course
5. Add conversations to that course
6. Delete the course
7. Look into the database, if the conversations have been removed

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
